### PR TITLE
Added WordStateMachine funtion. Josh is God

### PR DESCRIPTION
**Pull request by harshavarmak**
#DO NOT MERGE

### CODE REVIEW

**Comment by harshavarmak** [comment]
Need to check for integer value range.
Need to check for identifier being 256 characters.
Need to add logic to mutate GenericWord to the right token type

File: Compiler/src/implementation/Scanner.cpp [hunk 1]
```diff
@@ -196,6 +200,72 @@ Token Scanner::consumeZeroToken()
...
+			FilePosition++;
+			ValidEndState = true;
+			continue;
+		}
+
+		char NextChar = FileContents[FilePosition + 1];
+
+		//If enot the last character in file, check the next char.
+		//Also keep making sure that Accumulator doesn't exceed more than 256 characters.
+		if (isalpha(NextChar) && Accumulator.length() < 257) {
```
Comment (justiceadamsUNI) [inline Compiler/src/implementation/Scanner.cpp]: So right now if the accum is > 257 we throw the same error as if they had an unexpected character. Should we move this completely towards the end. So like, we get to a valid stop state, then we check the length. If invalid throw custom error : something like "This identifier is too long!"? That way we have different identifiers. Same concept for integers, do we want a custom error message for those cases? I feel like it's valuable enough to do.

Comment (justiceadamsUNI) [reply Compiler/src/implementation/Scanner.cpp]: Also, I'll tackle these merge conflicts tonight when I'm done with class. Should be straightforward enough.

**Comment by harshavarmak** [comment]
@justiceadamsUNI So I had a look at the conflicts, they mainly are change in file name from my desktop to your desktop, and the master branch still has getNextToken() instead of next(). I'll still let you handle it unless you tell me too =)

File: Compiler/src/implementation/Scanner.cpp [hunk 2]
```diff
@@ -11,6 +11,13 @@ Scanner::Scanner(string FilePath)
 	FilePosition = 0;
 }
 
+Token Scanner::peek()
+{
+	int tempFilePosition = FilePosition;
+	next();
```
Comment (justiceadamsUNI) [inline Compiler/src/implementation/Scanner.cpp]: we want to return next() here, but I'll add it right now. gonna fix conflicts

**Comment by justiceadamsUNI** [comment]
I'm good with this. Once it gets your two approvals go ahead and merge it. (unless we decide to update it)

**Comment by justiceadamsUNI** [comment]
Approved. Waiting on @hilliardj now

**Pull request merged**